src/main/resources/assets/opencomputers/loot/mineos/installer.lua

```diff
@@ -0,0 +1,385 @@
+  print(" ")
+  for i = 1, #govno do
+    print(govno[i])
+  end
+  print(" ")
+  return
+else
+  print("Done, everything's good. Proceed to downloading.")
+  print(" ")
+end
+
+------------------------------------------------------------------------------------------
+
+local lang
+
+local applications
+
+local padColor = 0x262626
+local installerScale = 1
+
+local timing = 0.2
+
+-----------------------------СТАДИЯ ПОДГОТОВКИ-------------------------------------------
+
+local function request(url)
+  local success, response = pcall(component.internet.request, url)
+  if success then
+    local responseData = ""
+    while true do
+      local data, responseChunk = response.read() 
+      if data then
+        responseData = responseData .. data
+      else
+        if responseChunk then
+          return false, responseChunk
+        else
+          return responseData
+        end
+      end
+    end
+  else
+    return false, reason
+  end
+end
+
+--БЕЗОПАСНАЯ ЗАГРУЗОЧКА
+local function getFromGitHubSafely(url, path)
+  local success, reason = request(url)
+  if success then
+    fs.makeDirectory(fs.path(path) or "")
+    fs.remove(path)
+    local file = io.open(path, "w")
+    file:write(success)
+    file:close()
+    return success
+  else
+    io.stderr:write("Can't download \"" .. url .. "\"!\n")
+    return -1
+  end
+end
+
+local GitHubUserUrl = "https://raw.githubusercontent.com/"
+
+
+--------------------------------- Стадия стартовой загрузки всего необходимого ---------------------------------
+
+
+local preLoadApi = {
+  { paste = "IgorTimofeev/OpenComputers/master/lib/ECSAPI.lua", path = "lib/ECSAPI.lua" },
+  { paste = "IgorTimofeev/OpenComputers/master/lib/colorlib.lua", path = "lib/colorlib.lua" },
+  { paste = "IgorTimofeev/OpenComputers/master/lib/image.lua", path = "lib/image.lua" },
+  { paste = "IgorTimofeev/OpenComputers/master/lib/config.lua", path = "lib/config.lua" },
+  { paste = "IgorTimofeev/OpenComputers/master/MineOS/Icons/Languages.pic", path = "MineOS/System/OS/Icons/Languages.pic" },
+  { paste = "IgorTimofeev/OpenComputers/master/MineOS/Icons/OK.pic", path = "MineOS/System/OS/Icons/OK.pic" },
+  { paste = "IgorTimofeev/OpenComputers/master/MineOS/Icons/Downloading.pic", path = "MineOS/System/OS/Icons/Downloading.pic" },
+  { paste = "IgorTimofeev/OpenComputers/master/MineOS/Icons/OS_Logo.pic", path = "MineOS/System/OS/Icons/OS_Logo.pic" },
+}
+
+print("Downloading file list")
+applications = seri.unserialize(getFromGitHubSafely(GitHubUserUrl .. "IgorTimofeev/OpenComputers/master/Applications.txt", "MineOS/System/OS/Applications.txt"))
+print(" ")
+
+for i = 1, #preLoadApi do
+  print("Downloading \"" .. fs.name(preLoadApi[i].path) .. "\"")
+  getFromGitHubSafely(GitHubUserUrl .. preLoadApi[i].paste, preLoadApi[i].path)
+end
+
+print(" ")
+print("Initialization stage is complete, loading installer")
+print(" ")
+
+package.loaded.ecs = nil
+package.loaded.ECSAPI = nil
+_G.ecs = require("ECSAPI")
```

Comment:
I would say this is absolutely a valid lootdisk addition imo (assuming
other complaints don't exist).

On Wed, 8 Jun 2016, 8:30 PM Danil Krivonos <notifications@github.com> wrote:

> In src/main/resources/assets/opencomputers/loot/mineos/installer.lua
> <https://github.com/MightyPirates/OpenComputers/pull/1857#discussion_r66231456>
> :
>
> > +print("Downloading file list")
> > +applications = seri.unserialize(getFromGitHubSafely(GitHubUserUrl .. "IgorTimofeev/OpenComputers/master/Applications.txt", "MineOS/System/OS/Applications.txt"))
> > +print(" ")
> > +
> > +for i = 1, #preLoadApi do
> > +  print("Downloading \"" .. fs.name(preLoadApi[i].path) .. "\"")
> > +  getFromGitHubSafely(GitHubUserUrl .. preLoadApi[i].paste, preLoadApi[i].path)
> > +end
> > +
> > +print(" ")
> > +print("Initialization stage is complete, loading installer")
> > +print(" ")
> > +
> > +package.loaded.ecs = nil
> > +package.loaded.ECSAPI = nil
> > +_G.ecs = require("ECSAPI")
>
> Please don't use global table to store things. Make it local.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/MightyPirates/OpenComputers/pull/1857/files/54b6c0d433f087c48cdbe1aec4cc4947e8693140#r66231456>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AC1W3K8OMGlCecLAev_7uxmR1lxjG0-lks5qJpm8gaJpZM4IwuiA>
> .
>